molgenis-questionnaires/src/main/frontend/src/components/QuestionnaireChapter.vue

```diff
@@ -0,0 +1,318 @@
+                  :options="options"
+                  @valueChange="onValueChanged">
+                </form-component>
+              </div>
+
+              <div class="spinner-container text-muted d-flex flex-column justify-content-center align-items-center"
+                   v-else>
+                <i class="fa fa-spinner fa-spin fa-3x my-3"></i>
+              </div>
+            </div>
+
+            <div class="card-footer">
+              <div class="row">
+
+                <!-- Shows previous button based on chapter progress -->
+                <div class="col-6">
+                  <button
+                    type="button"
+                    v-if="showPreviousButton"
+                    @click="navigateToPreviousChapter"
+                    class="btn btn-outline-secondary float-left">
+                    {{ 'questionnaire_previous_chapter' | i18n }}
+                  </button>
+                </div>
+
+                <!-- Shows next / submit button based on chapter progress -->
+                <div class="col-6">
+                  <button
+                    type="button"
+                    v-if="showNextButton"
+                    @click="navigateToNextChapter"
+                    class="btn btn-primary float-right">
+                    {{ 'questionnaire_next_chapter' | i18n }}
+                  </button>
+
+                  <button class="btn btn-primary float-right" @click="submitQuestionnaire" v-else>
+                    {{ 'questionnaire_submit' | i18n }}
+                  </button>
+                </div>
+
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <!-- Renders chapter navigation list -->
+        <div class="col-3">
+          <chapter-list
+            :questionnaireId="questionnaireId"
+            :currentChapterId="chapterId"
+            :changesMade="changesMade"
+            :saving="saving">
+          </chapter-list>
+        </div>
+
+      </div>
+    </template>
+
+  </div>
+</template>
+
+<script>
+  import ChapterList from './ChapterList'
+  import moment from 'moment'
+
+  import { FormComponent } from '@molgenis/molgenis-ui-form'
+  import 'flatpickr/dist/flatpickr.css'
+
+  export default {
+    name: 'QuestionnaireChapter',
+    props: {
+      chapterId: {
+        type: Number
+      },
+      questionnaireId: {
+        type: String
+      }
+    },
+    data () {
+      return {
+        changesMade: false,
+        formState: {},
+        loading: true,
+        navigationBlocked: false,
+        options: {
+          showEyeButton: false
+        },
+        saving: false,
+        showForm: true
+      }
+    },
+    methods: {
+      onValueChanged (formData) {
+        this.$store.commit('SET_FORM_DATA', formData)
+      },
+
+      /**
+       * Forces validation if a user wants to go to the next chapter
+       * Triggers client side validation by setting status to 'SUBMITTED'
+       */
+      navigateToNextChapter () {
```

Comment:
maybe change method name to indicate it is only the request to navigate and validation is done first